xdist/remote.py

```diff
@@ -181,8 +182,15 @@ def serialize_warning_message(warning_message):
     if isinstance(warning_message.message, Warning):
         message_module = type(warning_message.message).__module__
         message_class_name = type(warning_message.message).__name__
-        message_args = warning_message.message.args
         message_str = str(warning_message.message)
+        # check now if we can serialize the warning arguments (#349)
+        # if not, we will just use the exception message on the master node
+        try:
+            dumps(warning_message.message.args)
```

Comment:
something about ` message_args = None` sits wrong wit me but i have no idea yet what it is

Reply:
Ouch, yeah, it might be because I forgot to handle `message_args` being None on `master`. This worked as is because `cls(*data["message_args"])` when `data["message_args"]` is None will also raise a `TypeError`.

Changed so we explicitly check if `data["message_args"]` is None on `master` for better clarity.